numalogic/connectors/rds/db/mysql_fetcher.py

```diff
@@ -0,0 +1,136 @@
+        will be established with SSL/TLS. - If SSL/TLS is not enabled or configured,
+        the connection will be established without SSL/TLS. - The connection object is returned
+        for further use in executing queries on the database.
+
+        """
+        connection = None
+        if self.db_config.ssl and self.db_config.ssl_enabled:
+            connection = pymysql.connect(
+                host=self.db_config.endpoint,
+                port=self.db_config.port,
+                user=self.db_config.database_username,
+                password=self.get_password(),
+                db=self.db_config.database_name,
+                ssl=self.db_config.ssl.__dict__,
+                cursorclass=pymysql.cursors.DictCursor,
+                charset="utf8mb4",
+                connect_timeout=self.db_config.database_connection_timeout,
+            )
+        else:
+            connection = pymysql.connect(
+                host=self.db_config.endpoint,
+                port=self.db_config.port,
+                user=self.db_config.database_username,
+                password=self.get_password(),
+                db=self.db_config.database_name,
+                cursorclass=pymysql.cursors.DictCursor,
+                charset="utf8mb4",
+                connect_timeout=self.db_config.database_connection_timeout,
+            )
+
+        return connection
+
+    def get_db_cursor(self, connection):
+        """
+        Returns a cursor object for executing queries on the database.
+
+        Arguments:
+        - connection (pymysql.connections.Connection): The connection object for the
+        MySQL database.
+
+        Returns
+        -------
+            pymysql.cursors.Cursor: The cursor object for executing queries on the database.
+
+        Raises
+        ------
+            None
+
+        Notes
+        -----
+            - The cursor object is used to execute queries on the database.
+            - The connection object must be established before calling this method.
+
+        """
+        return connection.cursor()
+
+    def execute_query(self, query) -> pd.DataFrame:
+        """
+        Executes the given query on the database and returns the result as a pandas DataFrame.
+
+        Arguments:
+            query (str): The SQL query to be executed.
+
+        Returns
+        -------
+            pandas.DataFrame: The result of the query as a DataFrame.
+
+        Notes
+        -----
+            - This method establishes a connection to the MySQL database using the
+            provided configuration.
+            - It retrieves a cursor object for executing queries on the database.
+            - The query is executed using the cursor object.
+            - The result is fetched and converted into a DataFrame.
+            - The execution time of the query is logged using the _LOGGER object.
+        """
+        _start_time = time.perf_counter()
+        connection = self.get_connection()
```

Comment:
Does this create a new connection everytime this is called?

Reply:
Yes, it create new connection everytime and we handle pooling at RDS proxy level.